src/main/java/ca/team2706/frc/robot/sensors/AnalogSelector.java

```diff
@@ -25,28 +25,38 @@ public void initSendable(SendableBuilder builder) {
     }
 
     public int getIndex() {
-        // Check each voltage range
-        for (int i = 0; i < voltages.length; i++) {
-            // Get the current voltage
-            double voltage = analogInput.getAverageVoltage();
 
+        final double voltage = analogInput.getAverageVoltage();
+
+        int index = 0;
+        // Check each voltage range
+        for (int i = 0; i < VOLTAGE_RANGES.length; i++) {
             // Check if the voltage is within the current voltage range
-            if (voltage >= voltages[i].min && voltage < voltages[i].max) {
-                // The selector is within this range
-                return i;
+            if (VOLTAGE_RANGES[i].isWithin(voltage)) {
+                index = i;
+                break;
             }
         }
 
-        // Default to index of 0
-        return 0;
+        return index;
     }
 
-    private static class Range {
+    public static class Range {
         public final double min, max;
 
-        Range(double min, double max) {
+        public Range(double min, double max) {
             this.min = min;
             this.max = max;
         }
+
+        /**
+         * Determines if the number is within this range.
+         *
+         * @param number The number to be tested.
+         * @return True if it's within range, false otherwise.
+         */
+        public boolean isWithin(final double number) {
+            return min <= number && number <= max;
```

Comment:
To be consistent with the old code, this should be `min <= number < max`, not `min <= number <= max`